File: regular_assignments/day_008/count_of_subsequence_in_given_string/solution.rb
class Solution
  def call
    solve("ABCGAG") # 3
    solve("GAB") # 0
  end

  def solve(str)
    count_a = 0
    ans = 0
    i = 0
    while i < str.length
      if str[i] == 'A'
        count_a += 1
      elsif str[i] == 'G'
        ans += count_a
      end
      i += 1
    end
    ans
  end
end

# puts '-' * 100
# puts Solution.new.solve("ABCGAG")
# puts '-' * 100
# puts Solution.new.solve("GAB")

File: regular_assignments/day_008/count_of_subsequence_in_given_string/solution.go
package countOfSubsequenceInString

func Call() {
	solve("ABCGAG")
	solve("GAB")
}

func solve(str string) int {
	countA := 0
	ans := 0
	for i := 0; i < len(str); i++ {
		if str[i] == 'A' {
			countA++
		} else if str[i] == 'G' {
			ans += countA
		}
	}
	return ans
}
